pocs/utils/__init__.py

```diff
@@ -135,12 +135,30 @@ def time_left(self):
                 self.restart()
                 return self.duration
             else:
-                return max(0, delta)
+                return max(0.0, delta)
 
     def restart(self):
         """Restart the timed duration."""
         self.target_time = time.monotonic() + self.duration
 
+    def sleep(self, max_sleep=None):
```

Comment:
For the usage that you have here, with very small times, I don't think this is a problem but it concerns me that this could be blocking on long countdown, such as used in exposures. The answer of course is to not use this method then but I can see it being temping. 

Maybe a note in the docstring about the (obvious) fact that it is blocking. A callback would be nice that's probably overkill.

Reply:
Created issue #686 to track this.